### PR TITLE
#128 Allow node/breaker networks

### PR DESCRIPTION
**Pull request by zamarrenolm**

### CODE REVIEW

File: sources/Inputs/src/NetworkManager.cpp [hunk 1]
```diff
@@ -57,9 +57,10 @@ NetworkManager::buildTree() {
   const auto& voltageLevels = network->getVoltageLevels();
   for (auto it_v = voltageLevels.begin(); it_v != voltageLevels.end(); ++it_v) {
     if ((*it_v)->getVoltageLevelTopologyKind() == DYN::VoltageLevelInterface::NODE_BREAKER) {
-      LOG(error) << MESS(NodeBreakerInterface, (*it_v)->getID()) << LOG_ENDL;
-      // This is done in class constructor so we exit directly
-      std::exit(EXIT_FAILURE);
+      // Do not emit error if a node/breaker topology is found
+      // A version of Dynawo that supports node/breaker is required
+      // See https://github.com/dynawo/dynawo/pull/1536
```
Comment (rosiereflo) [inline sources/Inputs/src/NetworkManager.cpp:62]: We generally avoid referencing github issues in the code.
Did you test that the main connex component is correct in the case node_breaker?

**Comment by marcochiaramello** [comment]
some of tests need to be updated as failing

**Comment by zamarrenolm** [comment]
> some of tests need to be updated as failing

tests were failing due to testing environment having multiple versions of same IIDM shared libraries. It has been fixed trying to load only the current version. See this [commit](https://github.com/dynawo/dynawo/commit/4fbb2cf2c2c6fc76aa4322e3cdc714cfe3ac55aa)

File: sources/Inputs/src/NetworkManager.cpp (hunk 1 above)
Comment (zamarrenolm) [reply sources/Inputs/src/NetworkManager.cpp]: Reference removed.
What do you main by testing that main connex component is correct ?

Comment (rosiereflo) [reply sources/Inputs/src/NetworkManager.cpp]: thanks!
I mean that there is an algorithm lused ater in this code that go through the network to detect which nodes belong to the main component connex. I just wanted to make sure that it still works when node breaker nodes are involved.

**Comment by marcochiaramello** [comment]
Thanks for the answer! I think the following tests (MAIN.launch, MAIN.launch_P, MAIN.launch_infinite, MAIN.hvdc, MAIN.distant_regulation and MAIN.special_characters) are failing due to new version iidm (output iidm is now 1.3.2 against reference being 1.2.0). To be discussed!

**Comment by zamarrenolm** [comment]
> Thanks for the answer! I think the following tests (MAIN.launch, MAIN.launch_P, MAIN.launch_infinite, MAIN.hvdc, MAIN.distant_regulation and MAIN.special_characters) are failing due to new version iidm (output iidm is now 1.3.2 against reference being 1.2.0). To be discussed!

Sorry, I though you refer to the failing tests in dynawo. I did not run the dynaflow-launcher tests locally. Is there a reason they are not run in continuous integration ?

**Comment by rosiereflo** [comment]
> > Thanks for the answer! I think the following tests (MAIN.launch, MAIN.launch_P, MAIN.launch_infinite, MAIN.hvdc, MAIN.distant_regulation and MAIN.special_characters) are failing due to new version iidm (output iidm is now 1.3.2 against reference being 1.2.0). To be discussed!
> 
> Sorry, I though you refer to the failing tests in dynawo. I did not run the dynaflow-launcher tests locally. Is there a reason they are not run in continuous integration ?

Yes, continuous integration is not run on DynaFlow-Launcher as we did not find a good way to have the latest dynawo master available when compiling DLF. This is something that we need to solve at some point.
However there is a  nightly build that run those tests on a daily basis.

**Comment by zamarrenolm** [comment]
> > > Thanks for the answer! I think the following tests (MAIN.launch, MAIN.launch_P, MAIN.launch_infinite, MAIN.hvdc, MAIN.distant_regulation and MAIN.special_characters) are failing due to new version iidm (output iidm is now 1.3.2 against reference being 1.2.0). To be discussed!
> > 
> > 
> > Sorry, I though you refer to the failing tests in dynawo. I did not run the dynaflow-launcher tests locally. Is there a reason they are not run in continuous integration ?
> 
> Yes, continuous integration is not run on DynaFlow-Launcher as we did not find a good way to have the latest dynawo master available when compiling DLF. This is something that we need to solve at some point.
> However there is a nightly build that run those tests on a daily basis.

I understand, thanks for the clarification

**Review by marcochiaramello — APPROVED** [review]

**Comment by zamarrenolm** [comment]
This PR is closed. 
The `master` branch has evolved and required changes to support node/breaker will be published in a new PR from a clean branch.

**Pull request closed without merging**